src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -77,14 +74,155 @@ public void registered(SchedulerDriver driver, FrameworkID frameworkId, MasterIn
     }
     log.info("Registered framework frameworkId=" + frameworkId.getValue());
   }
-
+  
   @Override
   public void reregistered(SchedulerDriver driver, MasterInfo masterInfo) {
     log.info("Reregistered framework.");
   }
 
+  @Override
+  public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
+    log.info(String.format(
+        "Received status update for taskId=%s state=%s message='%s' stagingTasks.size=%d",
+        status.getTaskId().getValue(),
+        status.getState().toString(),
+        status.getMessage(),
+        liveState.getStagingTasksSize()));
+
+    if (!isStagingState(status)) {
+      liveState.removeStagingTask(status.getTaskId());
+    }
+
+    if (isTerminalState(status)) {
+      liveState.removeTask(status.getTaskId());
+    } else if (isRunningState(status)) {
+      liveState.updateTaskForStatus(status); // TODO(rubbish) could terminal/running get pushed to
```

Comment:
I agree with the TODO. Could either push the other liveState update handling in here, or rename updateTaskForStatus to indicate that it is for running task states